mavros/src/plugins/setpoint_global_position.cpp

```diff
@@ -0,0 +1,192 @@
+/**
+ * @brief SetpointGlobalPosition plugin
+ * @file setpoint_global_position.cpp
+ * @author Mohamed Abdelkader 2017 <mohamedashraf123@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2017, Mohamed Abdelkader.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros/setpoint_mixin.h>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <geometry_msgs/PoseStamped.h>
+#include <sensor_msgs/NavSatFix.h>
+#include <mavros_msgs/GlobalPositionTarget.h>
+
+#define M_DEG_TO_RAD 0.01745329251994
+#define CONSTANTS_RADIUS_OF_EARTH 6371000.0
+//#define DBL_EPSILON 2.2204460492503131E-16
```

Comment:
Deg-to-rad may be done by `<angles.h>` library (already used in some plugins).
And if you decide to use constant, please write `pi/180.0` instead of that magic value.